process/block/preprocess/gasComputation.go

```diff
@@ -170,6 +184,38 @@ func (gc *gasComputation) ComputeGasConsumedByTx(
 		return 0, 0, process.ErrNilTransaction
 	}
 
+	if !gc.flagGasComputeV2.IsSet() {
+		return gc.computeGasConsumedByTxV1(txSenderShardId, txReceiverShardId, txHandler)
+	}
+
+	moveBalanceConsumption := gc.economicsFee.ComputeGasLimit(txHandler)
+
+	_, txCrossShard := gc.txTypeHandler.ComputeTransactionType(txHandler)
+	isSCCall := txCrossShard == process.SCDeployment ||
+		txCrossShard == process.SCInvoking ||
+		txCrossShard == process.BuiltInFunctionCall
+	if isSCCall {
+		isCrossShardSCCall := txSenderShardId != txReceiverShardId &&
+			moveBalanceConsumption < txHandler.GetGasLimit() &&
+			txCrossShard != process.BuiltInFunctionCall
+		if isCrossShardSCCall {
+			gasConsumedByTxInSenderShard := moveBalanceConsumption
+			gasConsumedByTxInReceiverShard := txHandler.GetGasLimit() - moveBalanceConsumption
+
+			return gasConsumedByTxInSenderShard, gasConsumedByTxInReceiverShard, nil
+		}
+
+		return txHandler.GetGasLimit(), txHandler.GetGasLimit(), nil
+	}
+
+	return moveBalanceConsumption, moveBalanceConsumption, nil
+}
+
+func (gc *gasComputation) computeGasConsumedByTxV1(
+	txSenderShardId uint32,
+	txReceiverShardId uint32,
+	txHandler data.TransactionHandler,
+) (uint64, uint64, error) {
 	moveBalanceConsumption := gc.economicsFee.ComputeGasLimit(txHandler)
 
 	txTypeInShard, _ := gc.txTypeHandler.ComputeTransactionType(txHandler)
```

Comment:
txType instead txTypeInShard

Reply:
this was the old code. copy pasted. would leave as it is